File: backend/server_delta/server_delta_app/models/financial_transaction/financial_transaction_model.py
from django.db import models
from server_delta_app import models as server_delta_app_models, managers


class FinancialTransactionModel(server_delta_app_models.BaseModel):
    """
        Model that represent the customer debt registered in app
    """
    class Meta:
        db_table = "financial_transaction"

    value = models.DecimalField(max_digits=10, decimal_places=2)
    description = models.CharField(max_length=1000)
    date = models.DateField()
    type_transaction = models.CharField(max_length=100)
    payment_type = models.CharField(max_length=100)
    customer_dossier = models.ForeignKey(server_delta_app_models.CustomerDossierModel, on_delete=models.CASCADE, related_name='financial_transactions')

    objects = managers.FinancialTransactionManager()

File: frontend/web-delta/src/app/models/financial_transaction/financial_transaction.model.ts
import {BaseModel} from '../base/base.model';

export class FinancialTransactionModel extends BaseModel {

  private _value: Number;
  private _description: String;
  private _type_transaction: String;
  private _payment_type: String;
  private _date: Date;
  private _customer_dossier: Number;

    /**
     * Getter value
     * @return {Number}
     */
  public get value(): Number {
    return this._value;
  }

    /**
     * Getter description
     * @return {String}
     */
  public get description(): String {
    return this._description;
  }

    /**
     * Getter type_transaction
     * @return {String}
     */
  public get type_transaction(): String {
    return this._type_transaction;
  }

    /**
     * Getter payment_type
     * @return {String}
     */
  public get payment_type(): String {
    return this._payment_type;
  }

    /**
     * Getter date
     * @return {Date}
     */
  public get date(): Date {
    return this._date;
  }

    /**
     * Getter customer_dossier
     * @return {Number}
     */
  public get customer_dossier(): Number {
    return this._customer_dossier;
  }

    /**
     * Setter value
     * @param {Number} value
     */
  public set value(value: Number) {
    this._value = value;
  }

    /**
     * Setter description
     * @param {String} value
     */
  public set description(value: String) {
    this._description = value;
  }

    /**
     * Setter type_transaction
     * @param {String} value
     */
  public set type_transaction(value: String) {
    this._type_transaction = value;
  }

    /**
     * Setter payment_type
     * @param {String} value
     */
  public set payment_type(value: String) {
    this._payment_type = value;
  }

    /**
     * Setter date
     * @param {Date} value
     */
  public set date(value: Date) {
    this._date = value;
  }

    /**
     * Setter customer_dossier
     * @param {Number} value
     */
  public set customer_dossier(value: Number) {
    this._customer_dossier = value;
  }
}
